captura/views.py

```diff
@@ -224,7 +227,121 @@ def create_estudio(request):
+    """
+    tutores = ['Tutor', 'Madre', 'Padre']
+    if request.method == 'POST':
+        instance = get_object_or_404(Integrante, pk=id_integrante)
+        integrante_form = IntegranteForm(request.POST, instance=instance)
+        if integrante_form.is_valid():
+            integrante_form.save()
+            rol = integrante_form.cleaned_data['Rol']
+            if rol == 'Alumno':
+                try:
+                    alumno = instance.alumno
+                    alumno_form = AlumnoForm(request.POST, instance=alumno)
+                    if alumno_form.is_valid():
+                        alumno_form.save()
+                except ObjectDoesNotExist:
+                    escuela = Escuela.objects.all().first()
+                    Alumno.objects.create(integrante=instance, numero_sae='000', escuela=escuela)
+                    return redirect(reverse('captura:integrante',
+                                            kwargs={'id_integrante': id_integrante}))
+            elif rol in tutores:
+                try:
+                    tutor = instance.tutor
+                    tutor_form = TutorForm(request.POST, instance=tutor)
+                    if tutor_form.is_valid():
+                        tutor_form.save()
+                except ObjectDoesNotExist:
+                    Tutor.objects.create(relacion=integrante_form.cleaned_data['Rol'],
+                                         integrante=instance)
+                    return redirect(reverse('captura:integrante',
+                                            kwargs={'id_integrante': id_integrante}))
+            return redirect(reverse('captura:integrantes',
+                                    kwargs={'id_familia': integrante_form.instance.familia.pk}))
+    else:
+        forms = {}
+        integrante = Integrante.objects.get(pk=id_integrante)
+        rol_integrante = 'Ninguno'
+        rol_disabled = False
+        try:
+            alumno = integrante.alumno
+            forms['form_alumno'] = AlumnoForm(instance=alumno)
+            rol_integrante = 'Alumno'
+            rol_disabled = True
+        except ObjectDoesNotExist:
+            pass
+        try:
+            tutor = integrante.tutor
+            forms['form_tutor'] = TutorForm(instance=tutor)
+            rol_integrante = 'Tutor'
+            rol_disabled = True
+        except ObjectDoesNotExist:
+            pass
+
+        forms['integrante_form'] = IntegranteForm(instance=integrante,
+                                                  initial={'Rol': rol_integrante})
+        if rol_disabled:
+            forms['integrante_form'].fields['Rol'].widget = HiddenInput()
+        form_view = 'captura/integrante_form.html'
+        return render(request, 'captura/captura_base.html', {'form_view': form_view,
+                                                             'forms': forms,
+                                                             'integrante': integrante})
```

Comment:
(Optional) Think if there is a way to refactor this view without using so many try-except statements. This view could use a bit refactoring